src/common/compress/zst/compress.c

```diff
@@ -0,0 +1,207 @@
+    ZSTD_CStream *context;                                          // Compression context
+    int level;                                                      // Compression level
+    IoFilter *filter;                                               // Filter interface
+
+    bool inputSame;                                                 // Is the same input required on the next process call?
+    size_t inputOffset;                                             // Current offset in input buffer
+    bool flushing;                                                  // Is input complete and flushing in progress?
+} ZstCompress;
+
+/***********************************************************************************************************************************
+Render as string for logging
+***********************************************************************************************************************************/
+static String *
+zstCompressToLog(const ZstCompress *this)
+{
+    return strNewFmt(
+        "{level: %d, inputSame: %s, inputOffset: %zu, flushing: %s}", this->level, cvtBoolToConstZ(this->inputSame),
+        this->inputOffset, cvtBoolToConstZ(this->flushing));
+}
+
+#define FUNCTION_LOG_ZST_COMPRESS_TYPE                                                                                             \
+    ZstCompress *
+#define FUNCTION_LOG_ZST_COMPRESS_FORMAT(value, buffer, bufferSize)                                                                \
+    FUNCTION_LOG_STRING_OBJECT_FORMAT(value, zstCompressToLog, buffer, bufferSize)
+
+/***********************************************************************************************************************************
+Free compression context
+***********************************************************************************************************************************/
+OBJECT_DEFINE_FREE_RESOURCE_BEGIN(ZST_COMPRESS, LOG, logLevelTrace)
+{
+    ZSTD_freeCStream(this->context);
+}
+OBJECT_DEFINE_FREE_RESOURCE_END(LOG);
+
+/***********************************************************************************************************************************
+Compress data
+***********************************************************************************************************************************/
+static void
+zstCompressProcess(THIS_VOID, const Buffer *uncompressed, Buffer *compressed)
+{
+    THIS(ZstCompress);
+
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(ZST_COMPRESS, this);
+        FUNCTION_LOG_PARAM(BUFFER, uncompressed);
+        FUNCTION_LOG_PARAM(BUFFER, compressed);
+    FUNCTION_LOG_END();
+
+    ASSERT(this != NULL);
+    ASSERT(!(this->flushing && !this->inputSame));
+    ASSERT(this->context != NULL);
+    ASSERT(compressed != NULL);
+    ASSERT(!this->flushing || uncompressed == NULL);
+
+    // Initialize output buffer
+    ZSTD_outBuffer out = {.dst = bufRemainsPtr(compressed), .size = bufRemains(compressed)};
```

Comment:
I don't think we have a coding standard written for this, but I thought when we declare and initialize a structure that it would be on separate lines - like `ZSTD_inBuffer in` below (line 99 - 103)

Reply:
Usually I have been leaving them on one line if it fits, just like a function call. At least for new stuff. Since a lot of our designated initializers were converted from a list of initializations I did not necessarily reformat them as I went.

I'm not set on that, but if we decided to split them out we'll need to run through the code and make sure it is done everywhere.

Reply:
No problem. I'll make a note to update the coding standard that if it is not able to fit on one line then each structure member should be on a separate line. Sound ok?

Reply:
Sounds good to me.